ports/windows/mpconfigport.h

```diff
@@ -160,6 +160,22 @@ typedef int mp_int_t; // must be pointer size
 typedef unsigned int mp_uint_t; // must be pointer size
 #endif
 
+typedef long suseconds_t;
+
+// Something wrong in mingw's math.h leading to signbit et al. being defined
+// to versions taking float instead of double so fix this here.
+// Does mean this file must always come after #include <math.h>
+#if defined( __MINGW32__ ) && (MICROPY_FLOAT_IMPL == MICROPY_FLOAT_IMPL_DOUBLE)
+#undef signbit
+#undef isnan
+#undef isinf
+#undef fpclassify
+#define signbit __signbit
+#define isnan __isnan
+#define isinf(x) (__fpclassify(x) == FP_INFINITE)
+#define fpclassify __fpclassify
+#endif
```

Comment:
I'm not sure about this.  As you say there must be a better way to fix it.... does anyone else have problems with math.h under mingw?

Reply:
Don't worry about this too much, I'm thinking it has to be a build option or so and will figure that out later.